src/backend/src/api/endpoints/deployment.rs

```diff
@@ -379,27 +378,30 @@ impl DeploymentsEndpoints {
         }
 
         match update {
-            DeploymentState::Active => self.deployments_service.update_deployment_state(
-                calling_principal,
-                deployment_id,
-                update,
-                false,
-            ),
+            DeploymentState::Active => {}
             DeploymentState::FailedOnClient { .. } => {
-                self.deployments_service.update_deployment_state(
-                    calling_principal,
-                    deployment_id,
-                    update,
-                    false,
-                )?;
-
-                try_close_akash_deployment(&deployment_id).await
+                if let Err(e) = try_close_akash_deployment(&deployment_id).await {
+                    return self.deployments_service.set_failed_deployment(
+                        calling_principal,
+                        deployment_id,
+                        e.message().to_string(),
+                    );
+                }
+            }
+            _ => {
+                return Err(ApiError::invalid_argument(&format!(
+                    "Invalid update for deployment: {:?}",
+                    update
+                )))
             }
-            _ => Err(ApiError::invalid_argument(&format!(
-                "Invalid update for deployment: {:?}",
-                update
-            ))),
         }
+
+        self.deployments_service.update_deployment_state(
```

Comment:
I think it's better to put it back into the scope of `DeploymentState::Active`, no?

Otherwise at some point we might remove that branch thinking that is not needed and we would always end up executing the catch all branch

Reply:
Yeah, you're right! I just didn't want to repeat code in the first two branches, but in this case it may make sense